SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -51,29 +58,90 @@ QueryConstraint InterestFactory::CreateClientCheckoutRadiusConstraint(USpatialCl
+			ActorComponentSetToRadius);
 
-	// The previously built map dedupes spatial constraints. Now the actual query constraints can be built of the form:
-	// OR(AND(cyl(radius), OR(actor 1 components, actor 2 components, ...)), ...)
-	// which is equivalent to having a separate spatial query for each actor type if the radius is the same.
-	TArray<QueryConstraint> CheckoutRadiusConstraints = CheckoutRadiusConstraintUtils::BuildNonDefaultActorCheckoutConstraints(
-		DistanceToActorTypeComponents, ClassInfoManager);
+		// The previously built map dedupes spatial constraints. Now the actual query constraints can be built of the form:
+		// OR(AND(cyl(radius), OR(actor 1 components, actor 2 components, ...)), ...)
+		// which is equivalent to having a separate spatial query for each actor type if the radius is the same.
+		TArray<QueryConstraint> CheckoutRadiusConstraints = CheckoutRadiusConstraintUtils::BuildNonDefaultActorCheckoutConstraints(
+			DistanceToActorTypeComponents, ClassInfoManager);
 
-	// Add all the different actor queries to the overall checkout constraint.
-	for (auto& ActorCheckoutConstraint : CheckoutRadiusConstraints)
+		// Add all the different actor queries to the overall checkout constraint.
+		for (auto& ActorCheckoutConstraint : CheckoutRadiusConstraints)
+		{
+			CheckoutRadiusConstraintRoot.OrConstraint.Add(ActorCheckoutConstraint);
+		}
+	}
+	else
 	{
-		CheckoutRadiusConstraint.OrConstraint.Add(ActorCheckoutConstraint);
+		const TArray<float> NetCullDistances = ClassInfoManager->GetNetCullDistances();
+
+		for (float NetCullDistance : NetCullDistances)
+		{
+			const float MaxCheckoutRadiusMeters = CheckoutRadiusConstraintUtils::NetCullDistanceSquaredToSpatialDistance(NetCullDistance);
+			const uint32 NCDComponentId = ClassInfoManager->GetComponentIdForNetCullDistance(NetCullDistance);
+
+			QueryConstraint ComponentConstraint;
+			ComponentConstraint.ComponentConstraint = NCDComponentId;
+
+			if (SpatialGDKSettings->bEnableNetCullDistanceFrequency)
+			{
+				float FullFrequencyCheckoutRadius = MaxCheckoutRadiusMeters * SpatialGDKSettings->FullFrequencyNetCullDistanceRatio;
+
+				{
+					// Add default interest query which doesn't include a frequency
+					QueryConstraint RadiusConstraint;
+					RadiusConstraint.RelativeCylinderConstraint = RelativeCylinderConstraint{ FullFrequencyCheckoutRadius };
+
+					QueryConstraint CheckoutRadiusConstraint;
+					CheckoutRadiusConstraint.AndConstraint.Add(RadiusConstraint);
+					CheckoutRadiusConstraint.AndConstraint.Add(ComponentConstraint);
+
+					CheckoutRadiusConstraintRoot.OrConstraint.Add(CheckoutRadiusConstraint);
+				}
+
+				// Add interest query for specified distance/frequency pairs
+				for (auto DistanceFrequencyPair : SpatialGDKSettings->InterestRangeFrequencyPairs)
+				{
+					float CheckoutRadius = MaxCheckoutRadiusMeters * DistanceFrequencyPair.DistanceRatio;
```

Comment:
same as previous comment- why can't this be a radius directly?

Reply:
This was a decision made in the design of this feature. To ensure that frequency is relative to what the maximum checkout is per class type.